packages/tools/fetch-tool/src/fluidFetchMessages.ts

```diff
@@ -140,16 +194,28 @@ async function* saveOps(
     // Split into 100K ops
     const chunk = 100 * 1000;
 
+    let sequencedMessages: ISequencedDocumentMessage[] = [];
+    // current sequence number to start writing from
+    let curr = 1;
     // Figure out first file we want to write to
     let index = 0;
     if (files.length !== 0) {
-        index = (files.length - 1);
+        index = files.length - 1;
+        const name = filenameFromIndex(index);
+        const fileContent = fs.readFileSync(`${dir}/messages${name}.json`, { encoding: "utf-8" });
+        const messages: ISequencedDocumentMessage[] = JSON.parse(fileContent);
+        // set curr to be the number of the first op in the last file and add one.
+        curr = messages[0].sequenceNumber + 1;
     }
-    let curr = index * chunk + 1;
 
-    let sequencedMessages: ISequencedDocumentMessage[] = [];
     while (true) {
         const result: IteratorResult<ISequencedDocumentMessage[]> = await gen.next();
+        // if there was gap in ops, change curr to currSeq,
+        // and index should be 0 anyways because we ask user to delete files
+        curr = seqNumMismatch ? currSeq : curr;
+        // if there were exisintg files being read, change index so the new file has the correct name
+        index = readFromMessages ? Math.floor(curr / chunk) : index;
```

Comment:
Index should always be the last file index because we want to start storing ops from last file if it is present and 0 otherwise. So no need for variable readFromMessages either.